src/components/numeral-date/numeral-date.pw.tsx

```diff
@@ -0,0 +1,685 @@
+      <NumeralDateComponent
+        onChange={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await numeralDateInput(page, 0).type(inputValue);
+
+    expect(callbackCount).toBe(1);
+  });
+
+  test("should call onBlur callback when a blur event is triggered", async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <NumeralDateComponent
+        onBlur={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const input = numeralDateInput(page, 0);
+    await input.focus();
+    await input.blur();
+
+    await page.waitForTimeout(50);
+
+    await expect(input).not.toBeFocused();
+    expect(callbackCount).toBe(1);
+  });
+});
+
+test.describe("check accessibility for NumeralDate component", () => {
+  test("should pass accessibility tests for NumeralDate component", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<NumeralDateComponent />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component ", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<NumeralDateControlled />);
+
+    await checkAccessibility(page);
+  });
+
+  ([
+    ["dd", "mm", "yyyy"],
+    ["mm", "dd", "yyyy"],
+    ["dd", "mm"],
+    ["mm", "dd"],
+    ["mm", "yyyy"],
+  ] as NumeralDateProps["dateFormat"][]).forEach((dateFormat) => {
+    test(`should pass accessibility tests for NumeralDate when dateFormat prop is ${dateFormat}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<NumeralDateComponent dateFormat={dateFormat} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with enableInternalError prop", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<NumeralDateControlled enableInternalError />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with enableInternalWarning prop", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<NumeralDateControlled enableInternalWarning />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with labelInline prop", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <NumeralDateControlled labelInline labelAlign="right" labelWidth={30} />
+    );
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with adaptive behaviour", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <NumeralDateControlled
+        labelInline
+        labelAlign="right"
+        labelWidth={30}
+        adaptiveLabelBreakpoint={960}
+      />
+    );
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with labelHelp prop", async ({
```

Comment:
comment: You probably want to trigger the help tooltip here and then run the check